OpenRA.Mods.Common/Activities/Air/HeliFly.cs

```diff
@@ -53,6 +54,12 @@ public override Activity Tick(Actor self)
 			if (IsCanceled || !target.IsValidFor(self))
 				return NextActivity;
 
+			if (!playedSound && helicopter.Info.TakeoffSound != null && helicopter.CenterPosition.Z == helicopter.Info.LandAltitude.Length)
```

Comment:
This would be a lot clearer if it uses `LandAltitude.Z` instead of `LandAltitude.Length`, which is the same, because X and Y are supposed to be 0.